controller/src/beerocks/master/tasks/client_steering_task.cpp

```diff
@@ -249,11 +249,18 @@ void client_steering_task::steer_sta()
 void client_steering_task::handle_event(int event_type, void *obj)
 {
     if (event_type == STA_CONNECTED) {
-        TASK_LOG(DEBUG) << "steering successful for sta " << sta_mac;
-        steering_success = true;
+        auto current_bssid = database.get_node_parent(sta_mac);
+        if (target_bssid == current_bssid) {
+            TASK_LOG(DEBUG) << "steering successful for sta " << sta_mac << " to bssid "
+                            << current_bssid;
+            steering_success = true;
+        } else {
+            TASK_LOG(DEBUG) << "sta " << sta_mac << " steered to bssid " << current_bssid
```

Comment:
What happens in this case?
You don't mark the steering as successful, the task will retry to steer this station again?
As far as I remember, the task won't finish as long as `steering_success != true`

Reply:
The task finishes as soon as *a* DISCONNECT and *a* CONNECT event is received. In the FINALIZE state, the STA node is updated with update_node_failed_5ghz_steer_attempt if success is not true.